tests/Microsoft.Identity.Test.Unit/PublicApiTests/AuthenticationResultTests.cs

```diff
@@ -52,6 +55,111 @@ public void GetAuthorizationHeader()
+            Assert.IsNotNull(claimsPrincipal);
+            Assert.AreEqual(13, claimsPrincipal.Claims.Count());
+            var claims = claimsPrincipal.Claims;
+            IDictionary<string, string> claimsDictionary = new Dictionary<string, string>();
+            foreach (Claim claim in claims)
+            {
+                claimsDictionary.Add(claim.Type, claim.Value);
+            }
+
+            Assert.IsTrue(claimsDictionary.TryGetValue("some_claim", out string value));
+            Assert.AreEqual("value", value);
+        }
+
+        [TestMethod]
+        public void GetIdTokenClaimsWithNullIdTokenTest()
+        {
+            var ar = new AuthenticationResult(
+                "at",
+                false,
+                "uid",
+                DateTime.UtcNow,
+                DateTime.UtcNow,
+                "tid",
+                new Account("aid", "user", "env"),
+                null,
+                new[] { "scope" }, Guid.NewGuid(),
+                "SomeTokenType",
+                new AuthenticationResultMetadata(TokenSource.Cache));
+
+            Assert.IsNull(ar.GetIdTokenClaims());
+        }
+
+        [TestMethod]
+        public void GetIdTokenClaimsWithInvalidIdTokenStringTest()
+        {
+            var ar = new AuthenticationResult(
+                "at",
+                false,
+                "uid",
+                DateTime.UtcNow,
+                DateTime.UtcNow,
+                "tid",
+                new Account("aid", "user", "env"),
+                "bad_token",
+                new[] { "scope" }, Guid.NewGuid(),
+                "SomeTokenType",
+                new AuthenticationResultMetadata(TokenSource.Cache));
+
+            try
+            {
+                ar.GetIdTokenClaims();
+                Assert.Fail("Should have failed due to invalid token");
+            } 
+            catch (MsalClientException e)
+            {
+                Assert.IsNotNull(e);
+                Assert.AreEqual(MsalError.InvalidJwtError, e.ErrorCode);
+            }
+        }
+
+        [TestMethod]
+        public void GetIdTokenClaimsWithInvalidIdTokenTest()
+        {
+            var ar = new AuthenticationResult(
+                "at",
+                false,
+                "uid",
+                DateTime.UtcNow,
+                DateTime.UtcNow,
+                "tid",
+                new Account("aid", "user", "env"),
+                MockHelpers.CreateIdTokenWithInvalidJson(),
+                new[] { "scope" }, Guid.NewGuid(),
+                "SomeTokenType",
+                new AuthenticationResultMetadata(TokenSource.Cache));
+
+            try
+            {
+                ar.GetIdTokenClaims();
+                Assert.Fail("Should have failed due to invalid token");
+            }
+            catch (MsalClientException e)
+            {
+                Assert.IsNotNull(e);
+                Assert.AreEqual(MsalError.JsonParseError, e.ErrorCode);
+            }
+        }
```

Comment:
This can be refactored like https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/cc1ba2338f98153bc2bd28e00bf4c17e73058ae2/tests/Microsoft.Identity.Test.Unit/CoreTests/InstanceTests/InstanceProviderTests.cs#L106-L107